ansible-automation-platform-operator/tasks/main.yml

```diff
@@ -0,0 +1,115 @@
+    - name: Create AutomationController
+      kubernetes.core.k8s:
+        name: aap-test
+        namespace: "{{ target_namespace }}"
+        api_version: automationcontroller.ansible.com/v1beta1
+        kind: AutomationController
+        definition:
+          metadata:
+            labels:
+              opdev: opcap
+          spec:
+            create_preload_data: true
+            route_tls_termination_mechanism: Edge
+            garbage_collect_secrets: false
+            ingress_type: Route
+            loadbalancer_port: 80
+            no_log: false
+            image_pull_policy: IfNotPresent
+            projects_storage_size: 8Gi
+            auto_upgrade: true
+            task_privileged: false
+            projects_storage_access_mode: ReadWriteMany
+            set_self_labels: true
+            projects_persistence: false
+            replicas: 1
+            admin_user: admin
+            loadbalancer_protocol: http
+            nodeport_port: 30080
+        state: present
+    - name: Test AutomationController conditions type Running
+      kubernetes.core.k8s_info:
+        namespace: "{{ target_namespace }}"
+        name: aap-test
+        api_version: automationcontroller.ansible.com/v1beta1
+        kind: AutomationController
+        wait: true
+        wait_condition:
+          status: "True"
+          type: "Running"
+        wait_timeout: 300
+      register: automationcontroller
+      failed_when: (automationcontroller.resources[0].status.conditions | selectattr('type', 'search', 'Running') | list | last).status != "True"
+    - name: Test AutomationController conditions type Failure
+      kubernetes.core.k8s_info:
+        namespace: "{{ target_namespace }}"
+        name: aap-test
+        api_version: automationcontroller.ansible.com/v1beta1
+        kind: AutomationController
+        wait: true
+        wait_condition:
+          status: "False"
+          type: "Failure"
+        wait_timeout: 300
+      register: automationcontroller
+      failed_when: (automationcontroller.resources[0].status.conditions | selectattr('type', 'search', 'Failure') | list | last).status != "False"
+    - name: Test AutomationController conditions type Successful
+      kubernetes.core.k8s_info:
+        namespace: "{{ target_namespace }}"
+        name: aap-test
+        api_version: automationcontroller.ansible.com/v1beta1
+        kind: AutomationController
+        wait: true
+        wait_condition:
+          status: "True"
+          type: "Successful"
+        wait_timeout: 300
+      register: automationcontroller
+      failed_when: (automationcontroller.resources[0].status.conditions | selectattr('type', 'search', 'Failure') | list | last).status != "False"
+  rescue:
+    - name: Print when failed
+      ansible.builtin.debug:
+        msg: "AutomationController failed. Err: {{ ansible_failed_result }}"
+    - name: Register failed tests
+      ansible.builtin.set_fact:
+        test_result: "failed"
```

Comment:
Do we want to stick with "pass" or "fail"?

Not a fan of mixed tense. "pass"/"failed"